docs/arcadia-users-group/20240730-arcadia-pycolor/lesson.md

```diff
@@ -0,0 +1,225 @@
+
+`arcadia-pycolor` is a Python package that provides a set of color palettes and convenience functions to style plots so that they are compatible with Arcadia's style guide. This notebook provides a quick introduction to `arcadia_pycolor` how to use it to style `matplotlib` and `seaborn` plots so that they comply with the Arcadia style guide, largely following the [Quickstart Guide](https://github.com/Arcadia-Science/arcadia-pycolor/blob/main/docs/quickstart.md).
+
+We'll cover the following topics:
+
+1. Installing `arcadia-pycolor`
+2. Configuring `matplotlib` defaults
+3. Accessing and using colors, palettes, and gradients
+4. Using `arcadia-pycolor` to style plots
+
+
+## 1. Install the `arcadia-pycolor` package
+
+We've designed `arcadia-pycolor` to be distributed via PyPI. In a virtual environment of your choice, run the following command in your terminal:
+
+```bash
+pip install arcadia-pycolor
+```
+
+### Requirements
+`arcadia-pycolor` requires Python >=3.9 and has the following dependencies (which are detected and installed automatically when you install `arcadia-pycolor` using `pip`):
+- `matplotlib` (>=3.7,!=3.8.0)
+- `colorspacious` (>=1.1.2)
+
+Additionally, `arcadia-pycolor` checks whether you have the **"Suisse Int'l"** family of fonts installed on your system. If these fonts aren't installed, `arcadia-pycolor` will default to **"Arial"**. You can download the "Suisse Int'l" font family from [this Google Drive link](https://drive.google.com/drive/folders/1xMLKDiaFVVg81xdS03uqsggjzkYyMrqJ?usp=drive_link) (must be accessed using an `@arcadiascience.com` email).
+
+The package can then be imported in notebooks or scripts using the following command:
+
+
+```python
+import arcadia_pycolor as apc
+```
+
+## 2. Set the default `matplotlib` styles
+
+The package provides a function called `apc.mpl.setup` that sets the default `matplotlib` styles to match the Arcadia style guide. This function only needs to be called once, ideally at the beginning of the notebook or script. The styles it sets will automatically apply to all plots in the notebook or script.
+
+### What does `apc.mpl.setup` do?
+- Sets the default font to **"Suisse Int'l"** (or **"Arial"** if not installed)
+- Changes default axis, axis label, legend, and color cycler styles to match the Arcadia style
+- Registers Arcadia's colors, palettes, and gradients with `matplotlib`
+
+
+```python
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+apc.mpl.setup()
+
+def example_plot(title: str = "Example plot"):
+    plt.figure(figsize = (5, 3.5))
+    plt.plot([3, 1, 4, 1, 4], label = "group a")
+    plt.plot([2, 7, 1, 4, 2], label = "group b")
+
+    plt.xlabel("time")
+    plt.ylabel("value")
+
+    plt.legend(title = "category")
+    plt.title(title)
+
+# Plot using default rcParams
+with mpl.rc_context(mpl.rcParamsDefault):
+    example_plot("Using default rcParams")
+    plt.show()
+
+# Plot using apc.mpl.setup()
+example_plot("Using arcadia-pycolor")
+plt.show()
```

Comment:
It might be confusing that a `with` block is only used for the default settings. It might be simpler to move the `apc.mpl.setup` call so it's just before the second `example_plot` call to avoid the need for the with block.